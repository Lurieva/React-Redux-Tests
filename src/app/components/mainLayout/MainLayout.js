import React from 'react';
import PropTypes from 'prop-types';

import Footer from '../footer/Footer';

import './mainLayout.scss';

const MainLayout = ({ children }) => (
    <div className="main-layout">
        <div className="main-layout__content">
            {children}
        </div>
        <Footer />
    </div>
);

MainLayout.propTypes = {
    children: PropTypes.element.isRequired
};

export default MainLayout;