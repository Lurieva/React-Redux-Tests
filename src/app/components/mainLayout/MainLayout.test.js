import React from 'react';
import { shallow } from 'enzyme';
import MainLayout from './MainLayout';

jest.mock('../footer/Footer', () => 'Footer');

describe('MainLayout', () => {
    it('should be defined', () => {
        expect(MainLayout).toBeDefined();
    });

    it('should render correctly', () => {
        const tree = shallow(
            <MainLayout />
        );

        expect(tree).toMatchSnapshot();
    });
});