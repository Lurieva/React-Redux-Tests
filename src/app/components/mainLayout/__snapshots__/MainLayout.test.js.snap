// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainLayout />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="main-layout__content"
        />,
        <Footer />,
      ],
      "className": "main-layout",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "main-layout__content",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "Footer",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="main-layout__content"
          />,
          <Footer />,
        ],
        "className": "main-layout",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "main-layout__content",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "Footer",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
